mixer/v1/config/client/auth.proto

```diff
@@ -0,0 +1,161 @@
+//   limitations under the License.
+
+syntax = "proto3";
+
+package istio.mixer.v1.config.client;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/duration.proto";
+
+option (gogoproto.goproto_getters_all) = false;
+option (gogoproto.equal_all) = false;
+option (gogoproto.gostring_all) = false;
+
+// APIKey defines configuration for generating the `api.key`
+// attribute from HTTP requests.
+//
+// See https://swagger.io/docs/specification/authentication/api-keys
+// for a general overview of API keys as defined by OpenAPI.
+message APIKey {
+  oneof key {
+    // API Key is sent as a query parameter. `query` represents the
+    // query string parameter name.
+    //
+    // For example, `query=api_key` should be used with the
+    // following request:
+    //
+    //     GET /something?api_key=abcdef12345
+    //
+    string query = 1;
+
+    // API key is send in a request header. `header` represents the
+    // header name.
+    //
+    // For example, `header=X-API-KEY` should be used with the
+    // following request:
+    //
+    //     GET /something HTTP/1.1
+    //     X-API-Key: abcdef12345
+    //
+    string header = 2;
+
+    // API key is sent in a
+    // [cookie](https://swagger.io/docs/specification/authentication/cookie-authentication),
+    //
+    // For example, `cookie=X-API-KEY` should be used for the
+    // following request:
+    //
+    //     GET /something HTTP/1.1
+    //     Cookie: X-API-KEY=abcdef12345
+    //
+    string cookie = 3;
+  }
+}
+
+message JWT {
+  // Identifies the principal that issued the JWT. See
+  // https://tools.ietf.org/html/draft-ietf-oauth-json-web-token-32#section-4.1.1
+  // Usually a URL or an email address.
+  //
+  // Example: https://securetoken.google.com
+  // Example: 1234567-compute@developer.gserviceaccount.com
+  //
+  string name = 1;
+
+  // The list of JWT
+  // [audiences](https://tools.ietf.org/html/draft-ietf-oauth-json-web-token-32#section-4.1.3).
+  // that are allowed to access. A JWT containing any of these
+  // audiences will be accepted.
+  //
+  // Any audience will be accepted if the audience list is empty.
+  //
+  // Example:
+  //
+  //     audiences:
+  //     - bookstore_android.apps.googleusercontent.com
+  //       bookstore_web.apps.googleusercontent.com
+  //
+  repeated string audiences = 2;
+
+  oneof public_key {
+    // URL of the provider's public key set to validate signature of the JWT. See
+    // [OpenID Discovery](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata).
+    // Optional if the key set document:
+    //  - can be retrieved from
+    //    [OpenID Discovery](https://openid.net/specs/openid-connect-discovery-1_0.html
+    //    of the issuer.
+    //  - can be inferred from the email domain of the issuer (e.g. a Google service account).
+    //
+    // Example: https://www.googleapis.com/oauth2/v1/certs
+    string jwks_uri = 3;
+
+    // Is this option needed?
+    string pem = 4;
```

Comment:
Is `pem` option needed? It looks like `jwks_uri` is required for https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata.

Reply:
Same question. What is the use case for pem option?

Reply:
I'd be happy to remove it. I only added it because the istio/proxy JWT auth filter had the option.